src/OpenTelemetry.Instrumentation.AspNet/Implementation/RequestMethodHelper.cs

```diff
@@ -0,0 +1,69 @@
+// <copyright file="RequestMethodHelper.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+
+namespace OpenTelemetry.Instrumentation.AspNet.Implementation;
+
+internal sealed class RequestMethodHelper
+{
+    private const string KnownHttpMethodsEnvironmentVariable = "OTEL_INSTRUMENTATION_HTTP_KNOWN_METHODS";
+
+    // List of known HTTP methods in order of expected frequency.
+    private readonly string[] knownHttpMethods = new[]
```

Comment:
why not use dictionary?

Reply:
Since we're on a hot code path I figured looping over a small set of items should be faster than using a dictionary, but I can change this back if necessary.

Since you asked I decided to do a small performance test and found this:
![image](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/assets/5044535/9b01ab36-3821-42fc-80f4-e10a85e747ac)

I can change it to a dictionary to match the other implementation you linked.